src/Data/Check.cs

```diff
@@ -21,23 +21,36 @@ public struct Location {
 
             //if there are requirements, loop through each requirement to see if any are fully met
             foreach (Dictionary<string, int> req in itemsRequired) {
-                //ensure req and items use same terms
-                if (SaveFile.GetInt("randomizer sword progression enabled") != 0) {
-                    if (req.ContainsKey("Stick")) {
-                        req["Sword Progression"] = 1;
-                        req.Remove("Stick");
-                    }
-                    if (req.ContainsKey("Sword")) {
-                        req["Sword Progression"] = 2;
-                        req.Remove("Sword");
-                    }
-                }
-
                 //check if this requirement is fully met, otherwise move to the next requirement
                 int met = 0;
                 foreach (string item in req.Keys) {
                     //TunicLogger.LogInfo(item);
+                    // don't need to check if ability shuffle is on since the abilities are precollected if ability shuffle is off
                     if (!inventory.ContainsKey(item)) {
+                        if (item == "Sword") {
+                            if (inventory.ContainsKey("Sword Upgrade") && inventory["Sword Upgrade"] >= 2) {
```

Comment:
Should this be `Sword Progression` instead of `Sword Upgrade`?

Reply:
woops